apps/web/src/app/(private)/settings/_components/ProviderApiKeys/New/index.tsx

```diff
@@ -51,44 +55,81 @@ export default function NewApiKey({
     >
       <form id='createApiKeyForm' action={action}>
         <FormWrapper>
-          <FormField label='Provider'>
-            <Select
-              required
-              name='provider'
-              onChange={(newValue) => setProviderType(newValue)}
-              defaultValue={data?.provider}
-              options={PROVIDER_OPTIONS}
-            />
-          </FormField>
-          <FormField label='Name'>
+          <FormField
+            label='Provider'
+            description="This is the name you'll use in the prompt editor to refer to use this provider and model."
+          >
             <Input
               required
               type='text'
               name='name'
-              defaultValue={data?.name}
-              placeholder='My API Key'
+              defaultValue={result?.name}
+              placeholder='My Provider'
+              // @ts-expect-error
+              errors={errors?.name}
             />
           </FormField>
-          <FormField label='Token'>
+          <FormField label='Source'>
+            <Select
+              required
+              name='provider'
+              onChange={(value) => setProvider(value as Providers)}
+              defaultValue={result?.provider}
+              options={PROVIDER_OPTIONS}
+              // @ts-expect-error
+              errors={errors?.provider}
+            />
+          </FormField>
+          <FormField label='API Key'>
```

Comment:
Same. You can remove all `FormField` components here

Reply:
Ok